Update GitHub Actions workflows

axunonb opened the pull request:
`publish.yml`:
- Triggered manually
- Explicitly checkes out the main branch and resets to specified tag

`daily_build.yml`:
- Triggered manually
- Builds and publishes current state of main branch to GitHub Packages.
- Daily build version set to '5.0.0-daily.##' with commit hash included as release notes.

axunonb committed:
Update GitHub Actions workflows

`publish.yml`:
- Triggered manually
- Explicitly checkes out the main branch and resets to specified tag

`daily_build.yml`:
- Triggered manually
- Builds and publishes current state of main branch to GitHub Packages.
- Daily build version set to '5.0.0-daily.##' with commit hash included as release notes.

axunonb commented on Ical.Net.Tests/RecurrenceTests.cs:
```diff
@@ -3201,7 +3201,7 @@ public void OccurrenceMustBeCompletelyContainedWithinSearchRange()
     /// Find a solution for issue #120 or close forever
     /// </summary>
-    [Test, Ignore("No solution for issue #120 yet", Until = "2024-12-31")]
+    [Test, Ignore("No solution for issue #120 yet", Until = "2025-02-28")]
```
Shouldn't we remove this test which is "on hold" since 2016

minichma replied:
I think this has been fixed in the meantime, maybe with #659. After tweaking the test a little, it seems to succeed:

```cs
    /// <summary>
    /// Evaluate relevancy and validity of the request.
    /// Find a solution for issue #120 or close forever
    /// </summary>
    [Test]
    public void EventsWithShareUidsShouldGenerateASingleRecurrenceSet()
    {
        //https://github.com/rianjs/ical.net/issues/120 dated Sep 5, 2016
        const string ical = """
            BEGIN:VCALENDAR
            PRODID:-//Google Inc//Google Calendar 70.9054//EN
            VERSION:2.0
            CALSCALE:GREGORIAN
            METHOD:PUBLISH
            X-WR-CALNAME:Calendar 2
            X-WR-TIMEZONE:Europe/Bucharest
            BEGIN:VEVENT
            DTSTART;TZID=Europe/Bucharest:20160829T110000
            DTEND;TZID=Europe/Bucharest:20160829T163000
            RRULE:FREQ=DAILY
            DTSTAMP:20160901T104339Z
            UID:gknfcr66sb7rpangtprsthmpn8@google.com
            CREATED:20160901T104300Z
            DESCRIPTION:
            LAST-MODIFIED:20160901T104311Z
            LOCATION:
            SEQUENCE:1
            STATUS:CONFIRMED
            SUMMARY:testRScuAD
            TRANSP:OPAQUE
            END:VEVENT
            BEGIN:VEVENT
            DTSTART;TZID=Europe/Bucharest:20160901T163000
            DTEND;TZID=Europe/Bucharest:20160901T220000
            DTSTAMP:20160901T104339Z
            UID:gknfcr66sb7rpangtprsthmpn8@google.com
            RECURRENCE-ID;TZID=Europe/Bucharest:20160901T110000
            CREATED:20160901T104300Z
            DESCRIPTION:
            LAST-MODIFIED:20160901T104314Z
            LOCATION:
            SEQUENCE:2
            STATUS:CONFIRMED
            SUMMARY:testRScuAD
            TRANSP:OPAQUE
            END:VEVENT
            BEGIN:VEVENT
            DTSTART;TZID=Europe/Bucharest:20160903T070000
            DTEND;TZID=Europe/Bucharest:20160903T123000
            DTSTAMP:20160901T104339Z
            UID:gknfcr66sb7rpangtprsthmpn8@google.com
            RECURRENCE-ID;TZID=Europe/Bucharest:20160903T110000
            CREATED:20160901T104300Z
            DESCRIPTION:
            LAST-MODIFIED:20160901T104315Z
            LOCATION:
            SEQUENCE:2
            STATUS:CONFIRMED
            SUMMARY:testRScuAD
            TRANSP:OPAQUE
            END:VEVENT
            END:VCALENDAR
            """;

        var calendar = Calendar.Load(ical);

        //The API should be something like:
        var orderedOccurrences = calendar.GetOccurrences()
            .Take(10)
            .Select(o => o.Period)
            .ToList();

        var expectedSept1Start = new CalDateTime(DateTime.Parse("2016-09-01T16:30:00"), "Europe/Bucharest");
        var expectedSept1End = new CalDateTime(DateTime.Parse("2016-09-01T22:00:00"), "Europe/Bucharest");
        Assert.Multiple(() =>
        {
            Assert.That(orderedOccurrences[3].StartTime, Is.EqualTo(expectedSept1Start));
            Assert.That(orderedOccurrences[3].EndTime, Is.EqualTo(expectedSept1End));
        });

        var expectedSept3Start = new CalDateTime(DateTime.Parse("2016-09-03T07:00:00"), "Europe/Bucharest");
        var expectedSept3End = new CalDateTime(DateTime.Parse("2016-09-03T12:30:00"), "Europe/Bucharest");
        Assert.Multiple(() =>
        {
            Assert.That(orderedOccurrences[5].StartTime, Is.EqualTo(expectedSept3Start));
            Assert.That(orderedOccurrences[5].EndTime, Is.EqualTo(expectedSept3End));
        });
    }
```

axunonb committed:
Fix unit test

RecurrenceTests.EventsWithShareUidsShouldGenerateASingleRecurrenceSet
Thanks @minichma
